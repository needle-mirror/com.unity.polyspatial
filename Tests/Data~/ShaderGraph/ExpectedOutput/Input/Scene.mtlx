<?xml version ="1.0"?>
<materialx version = "1.38" colorspace = "lin_rec709">
    <realitykit_geometry_switch_cameraindex name = "_1" type = "float">
        <input name = "right" type = "float" value = "1" />
    </realitykit_geometry_switch_cameraindex>
    <dot name = "_2" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_3" />
    </dot>
    <dot name = "_3" type = "vector4">
        <input name = "in" type = "vector4" nodename = "unity_FogColor" />
    </dot>
    <constant name = "unity_FogColor" type = "vector4">
        <input name = "value" type = "vector4" value = "0,0,0,0" />
    </constant>
    <dot name = "_4" type = "float">
        <input name = "in" type = "float" nodename = "_5" />
    </dot>
    <ifequal name = "_5" type = "float">
        <input name = "value1" type = "float" nodename = "_6" />
        <input name = "value2" type = "float" value = "0" />
        <input name = "in1" type = "float" nodename = "_7" />
        <input name = "in2" type = "float" nodename = "_49" />
    </ifequal>
    <convert name = "_6" type = "float">
        <input name = "in" type = "boolean" nodename = "FOG_LINEAR" />
    </convert>
    <constant name = "FOG_LINEAR" type = "boolean">
        <input name = "value" type = "boolean" value = "false" />
    </constant>
    <ifequal name = "_7" type = "float">
        <input name = "value1" type = "float" nodename = "_8" />
        <input name = "value2" type = "float" value = "0" />
        <input name = "in1" type = "float" nodename = "_9" />
        <input name = "in2" type = "float" nodename = "_43" />
    </ifequal>
    <convert name = "_8" type = "float">
        <input name = "in" type = "boolean" nodename = "FOG_EXP" />
    </convert>
    <constant name = "FOG_EXP" type = "boolean">
        <input name = "value" type = "boolean" value = "false" />
    </constant>
    <ifequal name = "_9" type = "float">
        <input name = "value1" type = "float" nodename = "_10" />
        <input name = "value2" type = "float" value = "0" />
        <input name = "in1" type = "float" value = "0" />
        <input name = "in2" type = "float" nodename = "_11" />
    </ifequal>
    <convert name = "_10" type = "float">
        <input name = "in" type = "boolean" nodename = "FOG_EXP2" />
    </convert>
    <constant name = "FOG_EXP2" type = "boolean">
        <input name = "value" type = "boolean" value = "false" />
    </constant>
    <subtract name = "_11" type = "float">
        <input name = "in1" type = "float" value = "1" />
        <input name = "in2" type = "float" nodename = "_12" />
    </subtract>
    <clamp name = "_12" type = "float">
        <input name = "in" type = "float" nodename = "_13" />
    </clamp>
    <power name = "_13" type = "float">
        <input name = "in1" type = "float" value = "2" />
        <input name = "in2" type = "float" nodename = "_14" />
    </power>
    <multiply name = "_14" type = "float">
        <input name = "in1" type = "float" nodename = "_15" />
        <input name = "in2" type = "float" nodename = "_18" />
    </multiply>
    <multiply name = "_15" type = "float">
        <input name = "in1" type = "float" nodename = "_16" />
        <input name = "in2" type = "float" nodename = "_18" />
    </multiply>
    <subtract name = "_16" type = "float">
        <input name = "in2" type = "float" nodename = "_17" />
    </subtract>
    <swizzle name = "_17" type = "float">
        <input name = "in" type = "vector4" nodename = "unity_FogParams" />
        <input name = "channels" type = "string" value = "y" />
    </swizzle>
    <constant name = "unity_FogParams" type = "vector4">
        <input name = "value" type = "vector4" value = "0,0,0,0" />
    </constant>
    <max name = "_18" type = "float">
        <input name = "in1" type = "float" nodename = "_19" />
        <input name = "in2" type = "float" value = "0" />
    </max>
    <subtract name = "_19" type = "float">
        <input name = "in1" type = "float" nodename = "_20" />
        <input name = "in2" type = "float" nodename = "_36" />
    </subtract>
    <subtract name = "_20" type = "float">
        <input name = "in2" type = "float" nodename = "_21" />
    </subtract>
    <swizzle name = "_21" type = "float">
        <input name = "in" type = "vector4" nodename = "_22" />
        <input name = "channels" type = "string" value = "z" />
    </swizzle>
    <transformmatrix name = "_22" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_23" />
        <input name = "mat" type = "matrix44" nodename = "_34" />
    </transformmatrix>
    <transformmatrix name = "_23" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_24" />
        <input name = "mat" type = "matrix44" nodename = "_31" />
    </transformmatrix>
    <combine4 name = "_24" type = "vector4">
        <input name = "in1" type = "float" nodename = "_25" />
        <input name = "in2" type = "float" nodename = "_29" />
        <input name = "in3" type = "float" nodename = "_30" />
        <input name = "in4" type = "float" value = "1" />
    </combine4>
    <swizzle name = "_25" type = "float">
        <input name = "in" type = "vector3" nodename = "_26" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <dot name = "_26" type = "vector3">
        <input name = "in" type = "vector3" nodename = "_28" />
    </dot>
    <position name = "_27" type = "vector3">
        <input name = "space" type = "string" value = "object" />
    </position>
    <multiply name = "_28" type = "vector3">
        <input name = "in1" type = "vector3" nodename = "_27" />
        <input name = "in2" type = "vector3" value = "1,1,-1" />
    </multiply>
    <swizzle name = "_29" type = "float">
        <input name = "in" type = "vector3" nodename = "_26" />
        <input name = "channels" type = "string" value = "y" />
    </swizzle>
    <swizzle name = "_30" type = "float">
        <input name = "in" type = "vector3" nodename = "_26" />
        <input name = "channels" type = "string" value = "z" />
    </swizzle>
    <multiply name = "_31" type = "matrix44">
        <input name = "in1" type = "matrix44" nodename = "_32" />
        <input name = "in2" type = "matrix44" value = "1,0,0,0,0,1,0,0,0,0,-1,0,0,0,0,1" />
    </multiply>
    <multiply name = "_32" type = "matrix44">
        <input name = "in1" type = "matrix44" value = "1,0,0,0,0,1,0,0,0,0,-1,0,0,0,0,1" />
        <input name = "in2" type = "matrix44" nodename = "_33" />
    </multiply>
    <realitykit_surface_model_to_world name = "_33" type = "matrix44">
    </realitykit_surface_model_to_world>
    <multiply name = "_34" type = "matrix44">
        <input name = "in1" type = "matrix44" nodename = "_35" />
        <input name = "in2" type = "matrix44" value = "1,0,0,0,0,1,0,0,0,0,-1,0,0,0,0,1" />
    </multiply>
    <realitykit_surface_world_to_view name = "_35" type = "matrix44">
    </realitykit_surface_world_to_view>
    <divide name = "_36" type = "float">
        <input name = "in1" type = "float" nodename = "_37" />
        <input name = "in2" type = "float" nodename = "_41" />
    </divide>
    <swizzle name = "_37" type = "float">
        <input name = "in" type = "vector4" nodename = "_38" />
        <input name = "channels" type = "string" value = "w" />
    </swizzle>
    <transformmatrix name = "_38" type = "vector4">
        <input name = "in" type = "vector4" value = "0,0,1,0" />
        <input name = "mat" type = "matrix44" nodename = "_39" />
    </transformmatrix>
    <transpose name = "_39" type = "matrix44">
        <input name = "in" type = "matrix44" nodename = "_40" />
    </transpose>
    <realitykit_surface_view_to_projection name = "_40" type = "matrix44">
    </realitykit_surface_view_to_projection>
    <subtract name = "_41" type = "float">
        <input name = "in1" type = "float" nodename = "_42" />
        <input name = "in2" type = "float" value = "1" />
    </subtract>
    <swizzle name = "_42" type = "float">
        <input name = "in" type = "vector4" nodename = "_38" />
        <input name = "channels" type = "string" value = "z" />
    </swizzle>
    <subtract name = "_43" type = "float">
        <input name = "in1" type = "float" value = "1" />
        <input name = "in2" type = "float" nodename = "_44" />
    </subtract>
    <clamp name = "_44" type = "float">
        <input name = "in" type = "float" nodename = "_45" />
    </clamp>
    <power name = "_45" type = "float">
        <input name = "in1" type = "float" value = "2" />
        <input name = "in2" type = "float" nodename = "_46" />
    </power>
    <multiply name = "_46" type = "float">
        <input name = "in1" type = "float" nodename = "_47" />
        <input name = "in2" type = "float" nodename = "_18" />
    </multiply>
    <subtract name = "_47" type = "float">
        <input name = "in2" type = "float" nodename = "_48" />
    </subtract>
    <swizzle name = "_48" type = "float">
        <input name = "in" type = "vector4" nodename = "unity_FogParams" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <subtract name = "_49" type = "float">
        <input name = "in1" type = "float" value = "1" />
        <input name = "in2" type = "float" nodename = "_50" />
    </subtract>
    <clamp name = "_50" type = "float">
        <input name = "in" type = "float" nodename = "_51" />
    </clamp>
    <add name = "_51" type = "float">
        <input name = "in1" type = "float" nodename = "_52" />
        <input name = "in2" type = "float" nodename = "_54" />
    </add>
    <multiply name = "_52" type = "float">
        <input name = "in1" type = "float" nodename = "_18" />
        <input name = "in2" type = "float" nodename = "_53" />
    </multiply>
    <swizzle name = "_53" type = "float">
        <input name = "in" type = "vector4" nodename = "unity_FogParams" />
        <input name = "channels" type = "string" value = "z" />
    </swizzle>
    <swizzle name = "_54" type = "float">
        <input name = "in" type = "vector4" nodename = "unity_FogParams" />
        <input name = "channels" type = "string" value = "w" />
    </swizzle>
    <multiply name = "_55" type = "vector4">
        <input name = "in1" type = "vector4" nodename = "_2" />
        <input name = "in2" type = "vector4" nodename = "_59" />
    </multiply>
    <add name = "_56" type = "vector4">
        <input name = "in1" type = "vector4" nodename = "_60" />
        <input name = "in2" type = "vector4" nodename = "_55" />
    </add>
    <GeometryModification name = "_57" type = "vertexshader">
    </GeometryModification>
    <realitykit_unlit name = "_58" type = "surfaceshader">
        <input name = "applyPostProcessToneMap" type = "boolean" value = "false" />
        <input name = "color" type = "color3" nodename = "_61" />
    </realitykit_unlit>
    <surfacematerial name = "USD_Scene" type = "material">
        <input name = "vertexshader" type = "vertexshader" nodename = "_57" />
        <input name = "surfaceshader" type = "surfaceshader" nodename = "_58" />
    </surfacematerial>
    <swizzle name = "_59" type = "vector4">
        <input name = "in" type = "float" nodename = "_4" />
        <input name = "channels" type = "string" value = "xxxx" />
    </swizzle>
    <swizzle name = "_60" type = "vector4">
        <input name = "in" type = "float" nodename = "_1" />
        <input name = "channels" type = "string" value = "xxxx" />
    </swizzle>
    <swizzle name = "_61" type = "color3">
        <input name = "in" type = "vector4" nodename = "_56" />
        <input name = "channels" type = "string" value = "xyz" />
    </swizzle>
</materialx>
