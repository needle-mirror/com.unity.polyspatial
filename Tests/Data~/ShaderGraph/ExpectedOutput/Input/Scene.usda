#usda 1.0
(
    metersPerUnit = 1
    upAxis = "Y"
)

def "MaterialX"
{
    def "Materials"
    {
        def Material "USD_Scene"
        {
            float4 inputs:unity_FogColor = (0, 0, 0, 0)
            bool inputs:FOG_LINEAR = 0
            bool inputs:FOG_EXP = 0
            bool inputs:FOG_EXP2 = 0
            float4 inputs:unity_FogParams = (0, 0, 0, 0)

            token outputs:mtlx:surface.connect = <realitykit_unlit.outputs:out>
            token outputs:realitykit:vertex.connect = <GeometryModifier.outputs:out>

            def Shader "realitykit_unlit"
            {
                uniform token info:id = "ND_realitykit_unlit_surfaceshader"
                bool inputs:applyPostProcessToneMap = 0
                color3f inputs:color.connect = <../_61.outputs:out>
                token outputs:out
            }

            def Shader "GeometryModifier"
            {
                uniform token info:id = "ND_realitykit_geometrymodifier_vertexshader"
                token outputs:out
            }

            def Shader "_61"
            {
                uniform token info:id = "ND_swizzle_vector4_color3"
                float4 inputs:in.connect = <../_56.outputs:out>
                string inputs:channels = "xyz"
                color3f outputs:out
            }

            def Shader "_56"
            {
                uniform token info:id = "ND_add_vector4"
                float4 inputs:in1.connect = <../_60.outputs:out>
                float4 inputs:in2.connect = <../_55.outputs:out>
                float4 outputs:out
            }

            def Shader "_60"
            {
                uniform token info:id = "ND_swizzle_float_vector4"
                float inputs:in.connect = <../_1.outputs:out>
                string inputs:channels = "xxxx"
                float4 outputs:out
            }

            def Shader "_1"
            {
                uniform token info:id = "ND_realitykit_geometry_switch_cameraindex_float"
                float inputs:right = 1
                float outputs:out
            }

            def Shader "_55"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1.connect = <../_2.outputs:out>
                float4 inputs:in2.connect = <../_59.outputs:out>
                float4 outputs:out
            }

            def Shader "_2"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_3.outputs:out>
                float4 outputs:out
            }

            def Shader "_3"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../.inputs:unity_FogColor>
                float4 outputs:out
            }

            def Shader "_59"
            {
                uniform token info:id = "ND_swizzle_float_vector4"
                float inputs:in.connect = <../_4.outputs:out>
                string inputs:channels = "xxxx"
                float4 outputs:out
            }

            def Shader "_4"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_5.outputs:out>
                float outputs:out
            }

            def Shader "_5"
            {
                uniform token info:id = "ND_ifequal_float"
                float inputs:value1.connect = <../_6.outputs:out>
                float inputs:value2 = 0
                float inputs:in1.connect = <../_7.outputs:out>
                float inputs:in2.connect = <../_49.outputs:out>
                float outputs:out
            }

            def Shader "_6"
            {
                uniform token info:id = "ND_convert_boolean_float"
                bool inputs:in.connect = <../.inputs:FOG_LINEAR>
                float outputs:out
            }

            def Shader "_7"
            {
                uniform token info:id = "ND_ifequal_float"
                float inputs:value1.connect = <../_8.outputs:out>
                float inputs:value2 = 0
                float inputs:in1.connect = <../_9.outputs:out>
                float inputs:in2.connect = <../_43.outputs:out>
                float outputs:out
            }

            def Shader "_9"
            {
                uniform token info:id = "ND_ifequal_float"
                float inputs:value1.connect = <../_10.outputs:out>
                float inputs:value2 = 0
                float inputs:in1 = 0
                float inputs:in2.connect = <../_11.outputs:out>
                float outputs:out
            }

            def Shader "_10"
            {
                uniform token info:id = "ND_convert_boolean_float"
                bool inputs:in.connect = <../.inputs:FOG_EXP2>
                float outputs:out
            }

            def Shader "_8"
            {
                uniform token info:id = "ND_convert_boolean_float"
                bool inputs:in.connect = <../.inputs:FOG_EXP>
                float outputs:out
            }

            def Shader "_49"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in1 = 1
                float inputs:in2.connect = <../_50.outputs:out>
                float outputs:out
            }

            def Shader "_50"
            {
                uniform token info:id = "ND_clamp_float"
                float inputs:in.connect = <../_51.outputs:out>
                float outputs:out
            }

            def Shader "_51"
            {
                uniform token info:id = "ND_add_float"
                float inputs:in1.connect = <../_52.outputs:out>
                float inputs:in2.connect = <../_54.outputs:out>
                float outputs:out
            }

            def Shader "_54"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                float4 inputs:in.connect = <../.inputs:unity_FogParams>
                string inputs:channels = "w"
                float outputs:out
            }

            def Shader "_52"
            {
                uniform token info:id = "ND_multiply_float"
                float inputs:in1.connect = <../_18.outputs:out>
                float inputs:in2.connect = <../_53.outputs:out>
                float outputs:out
            }

            def Shader "_53"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                float4 inputs:in.connect = <../.inputs:unity_FogParams>
                string inputs:channels = "z"
                float outputs:out
            }

            def Shader "_18"
            {
                uniform token info:id = "ND_max_float"
                float inputs:in1.connect = <../_19.outputs:out>
                float inputs:in2 = 0
                float outputs:out
            }

            def Shader "_19"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in1.connect = <../_20.outputs:out>
                float inputs:in2.connect = <../_36.outputs:out>
                float outputs:out
            }

            def Shader "_36"
            {
                uniform token info:id = "ND_divide_float"
                float inputs:in1.connect = <../_37.outputs:out>
                float inputs:in2.connect = <../_41.outputs:out>
                float outputs:out
            }

            def Shader "_41"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in1.connect = <../_42.outputs:out>
                float inputs:in2 = 1
                float outputs:out
            }

            def Shader "_42"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                float4 inputs:in.connect = <../_38.outputs:out>
                string inputs:channels = "z"
                float outputs:out
            }

            def Shader "_38"
            {
                uniform token info:id = "ND_transformmatrix_vector4"
                float4 inputs:in = (0, 0, 1, 0)
                matrix4d inputs:mat.connect = <../_39.outputs:out>
                float4 outputs:out
            }

            def Shader "_39"
            {
                uniform token info:id = "ND_transpose_matrix44"
                matrix4d inputs:in.connect = <../_40.outputs:viewToProjection>
                matrix4d outputs:out
            }

            def Shader "_40"
            {
                uniform token info:id = "ND_realitykit_surface_view_to_projection"
                matrix4d outputs:viewToProjection
            }

            def Shader "_37"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                float4 inputs:in.connect = <../_38.outputs:out>
                string inputs:channels = "w"
                float outputs:out
            }

            def Shader "_20"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in2.connect = <../_21.outputs:out>
                float outputs:out
            }

            def Shader "_21"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                float4 inputs:in.connect = <../_22.outputs:out>
                string inputs:channels = "z"
                float outputs:out
            }

            def Shader "_22"
            {
                uniform token info:id = "ND_transformmatrix_vector4"
                float4 inputs:in.connect = <../_23.outputs:out>
                matrix4d inputs:mat.connect = <../_34.outputs:out>
                float4 outputs:out
            }

            def Shader "_34"
            {
                uniform token info:id = "ND_multiply_matrix44"
                matrix4d inputs:in1.connect = <../_35.outputs:worldToView>
                matrix4d inputs:in2 = ((1, 0, 0, 0), (0, 1, 0, 0), (0, 0, -1, 0), (0, 0, 0, 1))
                matrix4d outputs:out
            }

            def Shader "_35"
            {
                uniform token info:id = "ND_realitykit_surface_world_to_view"
                matrix4d outputs:worldToView
            }

            def Shader "_23"
            {
                uniform token info:id = "ND_transformmatrix_vector4"
                float4 inputs:in.connect = <../_24.outputs:out>
                matrix4d inputs:mat.connect = <../_31.outputs:out>
                float4 outputs:out
            }

            def Shader "_31"
            {
                uniform token info:id = "ND_multiply_matrix44"
                matrix4d inputs:in1.connect = <../_32.outputs:out>
                matrix4d inputs:in2 = ((1, 0, 0, 0), (0, 1, 0, 0), (0, 0, -1, 0), (0, 0, 0, 1))
                matrix4d outputs:out
            }

            def Shader "_32"
            {
                uniform token info:id = "ND_multiply_matrix44"
                matrix4d inputs:in1 = ((1, 0, 0, 0), (0, 1, 0, 0), (0, 0, -1, 0), (0, 0, 0, 1))
                matrix4d inputs:in2.connect = <../_33.outputs:modelToWorld>
                matrix4d outputs:out
            }

            def Shader "_33"
            {
                uniform token info:id = "ND_realitykit_surface_model_to_world"
                matrix4d outputs:modelToWorld
            }

            def Shader "_24"
            {
                uniform token info:id = "ND_combine4_vector4"
                float inputs:in1.connect = <../_25.outputs:out>
                float inputs:in2.connect = <../_29.outputs:out>
                float inputs:in3.connect = <../_30.outputs:out>
                float inputs:in4 = 1
                float4 outputs:out
            }

            def Shader "_30"
            {
                uniform token info:id = "ND_swizzle_vector3_float"
                float3 inputs:in.connect = <../_26.outputs:out>
                string inputs:channels = "z"
                float outputs:out
            }

            def Shader "_26"
            {
                uniform token info:id = "ND_dot_vector3"
                float3 inputs:in.connect = <../_28.outputs:out>
                float3 outputs:out
            }

            def Shader "_28"
            {
                uniform token info:id = "ND_multiply_vector3"
                float3 inputs:in1.connect = <../_27.outputs:out>
                float3 inputs:in2 = (1, 1, -1)
                float3 outputs:out
            }

            def Shader "_27"
            {
                uniform token info:id = "ND_position_vector3"
                string inputs:space = "object"
                float3 outputs:out
            }

            def Shader "_29"
            {
                uniform token info:id = "ND_swizzle_vector3_float"
                float3 inputs:in.connect = <../_26.outputs:out>
                string inputs:channels = "y"
                float outputs:out
            }

            def Shader "_25"
            {
                uniform token info:id = "ND_swizzle_vector3_float"
                float3 inputs:in.connect = <../_26.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_43"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in1 = 1
                float inputs:in2.connect = <../_44.outputs:out>
                float outputs:out
            }

            def Shader "_44"
            {
                uniform token info:id = "ND_clamp_float"
                float inputs:in.connect = <../_45.outputs:out>
                float outputs:out
            }

            def Shader "_45"
            {
                uniform token info:id = "ND_power_float"
                float inputs:in1 = 2
                float inputs:in2.connect = <../_46.outputs:out>
                float outputs:out
            }

            def Shader "_46"
            {
                uniform token info:id = "ND_multiply_float"
                float inputs:in1.connect = <../_47.outputs:out>
                float inputs:in2.connect = <../_18.outputs:out>
                float outputs:out
            }

            def Shader "_47"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in2.connect = <../_48.outputs:out>
                float outputs:out
            }

            def Shader "_48"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                float4 inputs:in.connect = <../.inputs:unity_FogParams>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_11"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in1 = 1
                float inputs:in2.connect = <../_12.outputs:out>
                float outputs:out
            }

            def Shader "_12"
            {
                uniform token info:id = "ND_clamp_float"
                float inputs:in.connect = <../_13.outputs:out>
                float outputs:out
            }

            def Shader "_13"
            {
                uniform token info:id = "ND_power_float"
                float inputs:in1 = 2
                float inputs:in2.connect = <../_14.outputs:out>
                float outputs:out
            }

            def Shader "_14"
            {
                uniform token info:id = "ND_multiply_float"
                float inputs:in1.connect = <../_15.outputs:out>
                float inputs:in2.connect = <../_18.outputs:out>
                float outputs:out
            }

            def Shader "_15"
            {
                uniform token info:id = "ND_multiply_float"
                float inputs:in1.connect = <../_16.outputs:out>
                float inputs:in2.connect = <../_18.outputs:out>
                float outputs:out
            }

            def Shader "_16"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in2.connect = <../_17.outputs:out>
                float outputs:out
            }

            def Shader "_17"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                float4 inputs:in.connect = <../.inputs:unity_FogParams>
                string inputs:channels = "y"
                float outputs:out
            }
        }
    }
}
