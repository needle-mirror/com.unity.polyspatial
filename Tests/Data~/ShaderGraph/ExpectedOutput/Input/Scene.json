{"properties":[{"name":"unity_FogColor","type":3},{"name":"FOG_LINEAR","type":0},{"name":"FOG_EXP","type":0},{"name":"FOG_EXP2","type":0},{"name":"unity_FogParams","type":3}]}